### PR TITLE
Changed how buttons on the VM list work

### PR DESCRIPTION
**Pull request by chalettu**
This PR is to fix how action buttons work on the VM list that is part of the Service Detail page.

### CODE REVIEW

**Comment by chalettu** [comment]
@miq-bot add_label wip

File: client/app/states/services/details/_details.sass [hunk 1]
```diff
@@ -6,3 +6,11 @@
   .order-details
     .list-view-pf
       margin-top: 0
+      
+.vm-action-button
+  padding: 2px 2px
+
+#serviceList
```
Comment (AllenBW) [inline client/app/states/services/details/_details.sass]: code climate should be yelling atcha for this, by id is a big no no now days

File: client/app/states/services/details/details.state.js [hunk 2]
```diff
@@ -133,21 +150,39 @@
...
+          title: $translate('Open a HTML5 console for this VM'),
+          actionFn: openConsole,
+        },
+        {
+          actionName: 'cockpitConsole',
+          title: $translate('Open Cockpit console for this VM'),
+          class: 'fa fa-plane btn btn-default',
+          actionFn: openConsole,
+        },
+ /*        {
```
Comment (AllenBW) [inline client/app/states/services/details/details.state.js]: gotta clean this stuffz

File: client/app/states/services/details/details.state.js [hunk 3]
```diff
@@ -118,8 +117,26 @@
...
+
+      switch (action.actionName) {
+        case 'htmlConsole':
+          if (item['supports_console?'] && item.power_state === 'on') {
+            buttonEnabled = true;
+          }
+          break;
+        case 'cockpitConsole':
+          if (item['supports_cockpit?'] && item.power_state === 'on') {
+            buttonEnabled = true;
```
Comment (AllenBW) [inline client/app/states/services/details/details.state.js]: return true here, allows you to clean up break and return at the end

File: client/app/states/services/details/details.state.js [hunk 4]
```diff
@@ -118,8 +117,26 @@
...
       vm.powerOperationSuspendState = PowerOperations.powerOperationSuspendState;
+      vm.vmMenuButtons = setupVmMenuButtons();
     }
+    function vmButtonEnabled(action, item) {
+      var buttonEnabled = true;
+
+      switch (action.actionName) {
+        case 'htmlConsole':
+          if (item['supports_console?'] && item.power_state === 'on') {
+            buttonEnabled = true;
```
Comment (AllenBW) [inline client/app/states/services/details/details.state.js]: same thing here, `return = true`

File: client/app/states/services/details/details.html [hunk 5]
```diff
@@ -201,7 +201,7 @@ <h2 translate>Virtual Machines ({{ ::vm.vms.length }})</h2>
         <p ng-if="vm.vms.length === 0" translate>
           This service has no associated VMs
         </p>
-        <div pf-list-view id="serviceList" config="vm.listConfig" items="vm.vms" ng-if="vm.vms.length > 0">
+        <div pf-list-view id="serviceList" config="vm.listConfig" enable-button-for-item-fn="vm.vmMenuButtons.buttonEnabledFn" action-buttons="vm.vmMenuButtons.actionButtons" items="vm.vms" ng-if="vm.vms.length > 0">
```
Comment (AllenBW) [inline client/app/states/services/details/details.html]: eeek might want to change the id, vmList or something

File: client/app/states/services/details/details.state.js [hunk 6]
```diff
@@ -133,21 +150,39 @@
...
+      vmMenuButtons.buttonEnabledFn = vmButtonEnabled;
+      var viewVirtualMachineAction = function(action, item) {
+        $state.go('vms.details', {vmId: item.id});
+      };
+
+      vmMenuButtons.actionButtons = [
+        {
+          actionName: 'htmlConsole',
+          class: 'fa fa-html5 btn btn-default',
+          title: $translate('Open a HTML5 console for this VM'),
```
Comment (AllenBW) [inline client/app/states/services/details/details.state.js]: rather than $translate, `__('Open a HTML5 console for this VM')`

Comment (AllenBW) [inline client/app/states/services/details/details.state.js]: (underscore underscore)

**Comment by chalettu** [comment]
@miq-bot remove_label wip

**Comment by chalettu** [comment]
@AllenBW , I tried what you had suggested and that only affects the outer containers margin.  The buttons inside the container didn't budge when trying that tweak.

**Comment by chalettu** [comment]
@miq-bot add_label technical_debt

**Comment by chalettu** [comment]
@miq-bot add_label technical debt

**Pull request merged**